src/main/scala/beam/sim/BeamSim.scala

```diff
@@ -248,6 +255,51 @@ class BeamSim @Inject()(
     out.close()
   }
 
+  def createSummaryStatsGraph(fileName: String, event: IterationEndsEvent): Unit = {
```

Comment:
To reduce on reprocessing the same logic, could you expect the parts pulled out of the `event` - since that part won't change even if the file does - controlerIO and iteration.

Reply:
Hi @JustinPihony , controlerIO is not changing with `event` but the iteration will change with the event . I have changed the code as per your comment . Please review it.

Reply:
LGTM